string request
---
string response
string reason